### ISSUE TITLE
1.2.2 appears to be buggy

### ISSUE THREAD

**Issue by rungta** [vaersaagod/matrixmate#14]
We just noticed that MatrixMate had stopped functioning across two of our projects. Tried rolling back to 1.2.1 and things were back to normal. Haven’t been able to dig in and locate the root cause yet, but thought I’d report it anyway.

**Comment by mmikkel** [comment]
Hi @rungta, thanks for reporting.

Yeah... that's not a lot to go on :) We're using 1.2.2 on several projects and aren't aware of any issues.

If you're able, the following info would be great:

– Craft version
– List of other plugins installed
– Any JS errors logged in your browser's dev console
– Your `matrixmate.php` config file

**Comment by rungta** [comment]
> Yeah... that's not a lot to go on :)

I know, sorry about that. Had to switch back quickly to restore functionality first.

- Craft version: 3.3.15
- List of other plugins installed: Agnostic Fetch 2.0.4, Cache Flag 1.0.3, Calendar Links 1.0.1, Child Me! 1.0.6, Colour Swatches 1.2.3, Control Panel CSS 2.2.0, Element Index Defaults 1.0.3, Environment Label 3.1.5, Feed Me 4.1.2, Fingerprint Assets 1.0.1, iCalendar 1.1.0, Imager v2.3.0, Image Resizer 2.0.6, LJ Dynamic Fields 3.0.10, MatrixMate 1.2.1, Minify 1.2.9, No-Cache 2.0.5, Obfuscator 1.0.3, oEmbed 1.1.8, Redactor 2.4.0, Relabel 1.3.4, Retcon 2.0.12, Router 1.0.0-beta.1, SEOMate 1.1.1, Single Cat 1.2.1, Smith 1.0.0, Store Hours 2.1.1.1, Super Table 2.3.0, Table Maker 2.0.1, Transcoder 1.2.5, Typed link field 1.0.19, Typogrify 1.1.18, Validateit 1.0.3
- No JS errors in the console. Only this message `Double-instantiating a menu button on an element garnish.js:220:20` which is probably unrelated and is spit out for 1.2.1 as well as 1.2.2
- No errors in the Yii Debug Toolbar logs

Here’s a simplified version of our `matrixmate.php` config. This works on 1.2.1 but not on 1.2.2:
```php
<?php

return [
    'fields' => [
        'nav' => [
            '*' => [
                'types' => [
                    'group' => [
                        'tabs' => [[
                            'label' => 'Heading',
                            'fields' => 'heading',
                        ]],
                        'defaultTabName' => 'Links',
                    ],
                ],
            ],
        ],
    ],
];

```

**Comment by mmikkel** [comment]
Thank you for the additional info @rungta. I'll try to reproduce the issue ASAP.

Just a sanity check; from looking at the above config snippet, you've got a Matrix field `nav` with a block type `group`, correct?

**Comment by rungta** [comment]
That’s right, @mmikkel.

**Comment by rungta** [comment]
Hello again @mmikkel!

So I ran into a similar issue this week where MatrixMate (and some of your other essential plugins like _CP Field Inspect_, _Child Me!_ etc.) suddenly stopped working, as if they had been uninstalled. I hadn’t updated any of them so I decided to investigate what in my code had changed and isolated it to calling `Craft::$app->user->isAdmin` in my module's `init()` method. 

Once I moved the `Craft::$app->user->isAdmin` statement inside an event handler (so that it wouldn’t be accessed in the module’s `init()` method then MatrixMate (and the other plugins) were back in action. Any idea why this may be happening?

**Comment by mmikkel** [comment]
Hi @rungta,

Yeah, that's a [known issue](https://github.com/mmikkel/CpFieldInspect-Craft#cogwheels-not-appearing) with Craft.

TL;DR – never call `Craft::$app->user->isAdmin` (or `Craft::$app->user->getIdentity()`) inside a module's `init()` method or constructor – it makes Craft behave all sorts of funny. Personally I've had good results by deferring those method calls to an [`EVENT_AFTER_LOAD_PLUGINS`](https://docs.craftcms.com/api/v3/craft-services-plugins.html#event-after-load-plugins).

**Comment by rungta** [comment]
Ah right. Should’ve checked the docs first! Thanks, and sorry for the trouble.

**Comment by mmikkel** [comment]
@rungta No trouble at all :)
